src/generators/genlua.ml

```diff
@@ -245,21 +245,34 @@ let mk_mr_box ctx e =
 	mk_lua_code ctx.com code [e] e.etype e.epos
 
 (* create a multi-return select call for given expr and field name *)
-let mk_mr_select com e name =
+let mk_mr_select com e ecall name =
 	let i =
-		match follow e.etype with
+		match follow ecall.etype with
 		| TInst (c,_) ->
 			index_of (fun f -> f.cf_name = name) c.cl_ordered_fields
 		| _ ->
 			assert false
 	in
 	if i == 0 then
-	    e
+            mk_lua_code com "{0}" [ecall] e.etype e.epos
 	else
 	    let code = Printf.sprintf "_G.select(%i, {0})" (i + 1) in
-	    mk_lua_code com code [e] e.etype e.epos
+	    mk_lua_code com code [ecall] e.etype e.epos
```

Comment:
Note to self:  copy these over to Main, technically they are the more correct type behavior.